client/src/featureform/serving.py

```diff
@@ -126,22 +137,88 @@ def _create_channel(self, host, insecure, cert_path):
             return secure_channel(host, cert_path)
 
     def training_set(self, name, variation, include_label_timestamp):
-        return Dataset(self._stub).from_stub(name, variation)
+        return Dataset(self._stub, resource_client=self._resource_client).from_stub(name, variation)
 
     def features(self, features, entities):
+        return FeatureServer(self._stub, self._resource_client, features, entities)
+
+
+class FeatureServer:
+    def __init__(self, stub, resource_client, features, entities):
+        self._stub = stub
+        self._resource_client = resource_client
+        self._features = features
+        self._entities = entities
+        self._feature_list = []
+        if self._all_ready():
+            self._get_features()
+    
+    def __getitem__(self, item):
+        self._feature_list[item]
+
+    def _get_features(self):
         req = serving_pb2.FeatureServeRequest()
-        for name, value in entities.items():
+        for name, value in self._entities.items():
             entity_proto = req.entities.add()
             entity_proto.name = name
             entity_proto.value = value
-        for (name, variation) in features:
+        for (name, variation) in self._features:
             feature_id = req.features.add()
             feature_id.name = name
             feature_id.version = variation
         resp = self._stub.FeatureServe(req)
-        return [parse_proto_value(val) for val in resp.values]
+        self._feature_list = [parse_proto_value(val) for val in resp.values]
+    
+    
+    def wait(self, timeout=None):
+        """Waits for all features to be ready before retrieving them.
 
+        **Examples**:
+        ``` py
+            client = ff.ServingClient(local=True)
+            fpf = client.features([("avg_transactions", "quickstart")], {"CustomerID": "C1410926"}).wait(timeout=10)
+            # Run features through model
+        ```
+        Args:
+            timeout int: the maximum number of seconds to wait before throwing an error
 
+        Returns:
+            self (FeatureServer object): an object that acts as a list containing every feature retrieved
+        """
+        for (name, variant) in self._features:
+            self._wait_for_feature(name, variant, timeout)
+        self._get_features()
+        return self
+
+    def _wait_for_feature(self, name, variant, timeout):
+        feature = self._resource_client.get_feature(name, variant, verbose=False)
```

Comment:
Having all this verbose=False is annoying and bad form. I really think that all the verbose logic should be moved into the CLI

Reply:
Should I get rid of all the automatic printing in get.py then? (format_rows, and such)

Reply:
I believe the purpose behind this was to print it when its called in the CLI. You should check if these calls are being used there and move the print statement out of the call itself